src/main/client/src/components/@pages/AchievementPage.tsx

```diff
@@ -20,23 +22,31 @@ export const AchievementPage: React.FC = (props) => {
   const toast = useToast()
   const { id } = useParams()
   const navigate = useNavigate()
+  const { throwError } = useServiceContext()
+
   if (!id) {
+    navigate('/')
     return null
   }
 
   const getAchievementDetails = () => {
-    axios.get<AchievementFullDetailsView>(`${API_BASE_URL}/api/achievement/submit/${id}`).then((res) => {
-      if (!res.data.achievement) {
-        navigate('/bucketlist')
-        toast({
-          title: 'Challange nem található',
-          description: 'Ilyen challange nem létezik vagy nincs jogosultságod hozzá.',
-          status: 'error',
-          isClosable: true
-        })
-      }
-      setAchDetails(res.data)
-    })
+    axios
+      .get<AchievementFullDetailsView>(`${API_BASE_URL}/api/achievement/submit/${id}`)
+      .then((res) => {
+        if (!res.data.achievement) {
+          navigate('/bucketlist')
+          toast({
+            title: 'Challange nem található',
+            description: 'Ilyen challange nem létezik vagy nincs jogosultságod hozzá.',
```

Comment:
I know it is not this PR's scope, but could you rename all **challange** texts to **challenge**? (Maybe use `ctrl+shift+f`)

Reply:
Write it under the Bucketlist UI issue.